### PR TITLE
Minor fixes + cleanup

### PR DESCRIPTION
**Pull request by dwillmer**
Firstly, thanks for such a great package!

This PR is just a few minor cleanup items:

- [X] Changed string comparisons from `if x is 'str'` to `if x == 'str'`; the former is an identity comparison, not equality. It happens to be true for some strings because of string interning, but `==` should always be used for normal comparisons.
- [X] Removed unused arguments from `_draw` function - `return_data` and others weren't used in the function body.
- [X] Removed unreachable code in `normalize` function (branch criteria could never be True).
- [X] Separated out the multiply-nested function calls in DataGeometry class for clarity.
- [X] Changed comparisons of the form`if type(x) is list` to `if isinstance(x, list)`; The former doesn't return True for subclasses, so isinstance should always be used.
- [X] Set unused loop variables to `_`.
- [X] Removed unused imports.
- [X] Ensured all imports are at the top of the file (except lazy / circular ones)
- [X] Ensure 2 blank lines above functions/classes (PEP8), the code looks a bit weird without this.
- [X] Fixed typo `repect` -> `respect`, was copy-pasted in multiple docstrings.
- [X] Removed redundant `pass` before error raise

### CODE REVIEW

**Comment by jeremymanning** [comment]
@dwillmer thanks for this pull request!

@andrewheusser this looks good to me— can you take a look as well and merge if it looks good to you too?

File: hypertools/tools/normalize.py [hunk 1]
```diff
@@ -52,21 +54,18 @@ def normalize(x, normalize='across', internal=False, format_data=True):
...
 
-        elif normalize=='within':
+        elif normalize == 'within':
             normalized_x = [np.array([zscore(i[:,j], i[:,j]) for j in range(i.shape[1])]).T for i in x]
 
-        elif normalize=='row':
+        elif normalize == 'row':
             normalized_x = [np.array([zscore(i[j,:], i[j,:]) for j in range(i.shape[0])]) for i in x]
 
-        elif normalize==False:
```
Comment (andrewheusser) [inline hypertools/tools/normalize.py]: Nice catch! Although we want to be able to bypass normalization and return unnormalized data if `normalize=None`, so i think this bit should stay, but read:
```
if not normalize:
    normalized_x = x
```
and then the docstring should also remove mention of False

**Comment by andrewheusser** [comment]
this looks great and thanks for carefully documenting the changes! just made one comment regarding the change to the normalize function, but once that is resolved i think we're good to go

**Comment by andrewheusser** [comment]
Ah, nevermind.. now that I'm looking at the full script I see we've already handled that case at the top. Merging!

**Pull request merged**